crates/starknet-server/test_data/rpc/declare_v1.json

```diff
@@ -1 +1 @@
7ouuj37LeK7uSyW5udJFP4+x+lemAt9bBjlJOiiQ2gWW0JuiAwOq42MZqA5jcjN+r7u8XhcPc/N42P8qHyoi5k8ul9V7umBTuSPTW/LCgXXfsrMzgjPnFc/X5yv+YkUEd/ucL+SkvrS5xHVP5Oe+Jpt6jgLldPUfGOXfuVasymjds7GQO/maL/WzS2hHpmXlwUFnoS1jZIJeocgS6xJWDzyWpHA4t6bZ4aFPVSFY1SqqyZFXW0O4quOXQDHktM7yzVh9i6VA1A4zlUHUMzu9rnTbV6Mp/ZktPRZuLrp4/f2v+8ub3188TydqPEBgMAoYEBZAyTNUnThHgBAEoCMUAIwEYZgQAgiilnHPK1SOCcLEl0Ssi5dXV382b2zdvr369Nl/cXr8ydQafoApRY8e2g4sYTESBaY/rxMG+UlJP1xV1tD/k2dXLl+azN69v3149uzVvrl9eP1PsJjkEnGLEMGEUIf+sLjeggIwTjkBd3tf9qv8/C68yyZl6/te9XNYILSt18zZsNQzH54pcWT2vqthF2GyZk6gmkxpZVi3uM0OEnEhHWWTlgSK6gCaxgbDKsBHNkpr5XZaMIutzvKFEqg0o0VNVhpUnzTUWCbjGlEY9m8pZoCkQCijLQ1dZmcDuZBLSXiPkKmz4uXJOv17dXpu+z8p3VRAajAoCDOWeMAIGRxgKNQQYkOzsqtb9voTmX65eP395/bagd+UliRqaOECCIU4oUi6TGmp8EtzATEDIDUULQrtT8tvLN//I71w5Z4CIwSkBiMOde5DzyeyhQ4ccdLDdFRtNGm/ihEMJbAxW7G6wCV6bO96JGvNNnYRUdrlpX20uraqTUOeh5WiGZwtzRxec00JFd6zKXV9+pj6raP5b+ic1KwT9uys968GNxrov8N3lgyioJtHiAQDWaqCKQTxpJtjMMIB3l2xyIqZayFw82bFT1Hz96sWtef0/168LQmaIOGBAYH07CsAqUyGCEb6rU76+G6+uP8lpCz7ko3xY6n0Alc1FP1DRPjRiw20GlR1Ex/HZRnJtRGW/Xt+av7x88+yv5uvfX/1SFAVAQjjgQiWrHFOKMACQYM4AwkTfD8S4sasdbPq/ffHq+ub26tVvBYGIijiIAJhhZCDDICoL5ggqWihjnAnE1F8KVUETSnQUpmSgk/Xrm5t8QjQVOv5iHGKFBpUyEpWZK/koZDCDcgiQQXAjKqLUtZQOA0FFCsUG4IDqO3JUFKbYFwxARRUwEIECqWwW8EbE3Fz/9+/Xr59tkwqkQtGj+sJUn1qHTImDYkUZR0IFrwbUP2lq9E1YrAlBt383X7z+85sCMrDqGQkslFUSqASiyEC4YXc3L359fXX7+9vrcnAgbihzJAAwpQoVniPEmPoi2O79y9Uvk5nz8fW9do8dRrDJjrZHsrCNTppEtCnJtJGK5cug2ejUnLVs/AZb47AgjrN1HXO6sbm9sJmJpprzeTu+U6Kz7uZ7QM66r47BE+unDfxsRNQmhDLC6A9FGVI6MbAtWFolDHFf/LaLKD0FJhdXsdSuM0AluuoOT6luGsIpKZ+W0JQriV7AlEtJF6ZVPL3g16o3udAOsy0DKZwm2Q+Ukp11CKZ0R03hlJJSW4DKl0c/kMqnpRsza3nWri2W24XWjRbh1NnTMJXurTtwZXtqiK6MoFqCV5FIesFXETEdGVshwpZRxb1ArJCslti+/fJi6s26RVbQR3d4itpviKJQFC1hJ8l0L4hJktCqwRSiY/XFHEcG1cxwgo6eNOa+bbyst+52DZp1R10iJ9ZJY/hsJNPGIlq+DHoCUoaO9s2peMCJamSXrco2yybMtJt9NAUMtOFBX7745e3V238EG2m27Woh2EAYE4oYJVBwSBkDlBEDUcC4oJRRKBDiiBADGtTYeU0pQVT+apK+EhgggwmCEQMQMr0XFBrGzlt7Xo7thbV40Dlshw4n1svg9yDGJdLGFsQc3vvYCPMI9x7mSLaNRe6b69fPzVfXNzd6y/PtG+0eCtyCXsMVgDODY4AJFgIwKAxOBSKYKSex+1petOP67fXV84J1RAC4AIbAmAFDryXrLd+Aowa+IOr0b29f3BasXiLKEVe9UYA4I4ACzathGEgwvDuzcuq+Ck5F3c5ewpuHZa47qo2S1az2Dt259TCZWW6OqaPtD2Wu+sBtWnuBlDYWT3a3+GA//tvobsduBoFYL9sHAdS4hyZjQFwgbQSbOaw3tu5apt1QCFn3itqRRUFM2uRizB1JaCPiDtv92/r20P0pWBW3c0KyGnNtgKJgbmhzUX9FMTmOvp7P3PlUxp31xfRkdXdfkjDh2glT2e5fxc50aQVRWzbWKzny53zQd56M3coH/6az6NqfJmf/tmm6xnG/apcIl9wIBoouJ6rZR8GNkF1PzWZJaMM/JVpt+frIxlwW3KLUErfF9yrF32Wy7W6jur3Wve2obvtpNTVSQLyxAi374ZonF3rJYBm86X2Ntoizk/z3fz85e+e/mmqeeXV5yrreL2b3eusL0CYVtk/i7QcvbFy3r1/RiM9jl1OVdwGTXbyL6WHNWl1dbL0unLfgjUruEa97hKLG7QtNYJ17IzlqjOdO7ipvzmdn/qvxJeK1++rQbQUdtOW3sq3VdFy80HHRnR0XjDkuiAs9F23iuWCh5/r27f8B/fXlzOt2AQA="}, "nonce": "0x0", "signature": ["0xacde09b7b4218575c154dbf9a0598c4e76f13dba493d048e0e79221960d353", "0x1872cdcabaa4c2746741f120a82ca09ced10cd46f62bc2e7c07599aff180f80"], "max_fee": "0x2af56b9ed300", "sender_address": "0xd47d2f1a72351f234dfc42a3f6b5cd6a16b32491daeca9bbb02e9e990e0e12", "version": "0x1", "type": "DECLARE"}
```

Comment:
why the nonce is changed

Reply:
I think it was wrong all this time, but I guess the sofar the balance was internally checked before the nonce. I mentioned this in the PR description (last bullet).

Reply:
if it's a first tx by a predeployed account (or just an attempted tx by a non-deployed account), I think nonce should be zero